libraries/ESP8266WiFi/src/ESP8266WiFiMulti.cpp

```diff
@@ -323,12 +323,18 @@ wl_status_t ESP8266WiFiMulti::connectWiFiMulti(uint32_t connectTimeoutMs)
     }
     DEBUG_WIFI_MULTI("\n");
 
+    // Create indices for AP connection failures
+    uint8_t connectSkipIndex[_APlist.size()];
+    memset(connectSkipIndex, 0, sizeof(connectSkipIndex));
+
     // Connect to known WiFi AP's sorted by RSSI
     for (int8_t i = 0; i < numNetworks; i++) {
         // Get network information
         WiFi.getNetworkInfo(known[i], ssid, encType, rssi, bssid, channel, hidden);
 
-        for (auto entry : _APlist) {
+        for (uint8_t j = 0; j < _APlist.size(); j++) {
+            auto entry = _APlist[j];
```

Comment:
The current code is making a local copy for entry, and this is doing the same. Not a big deal at this time, because _APList is a vector of WiFiAPEntry, which is a struct with 2 ptrs, but still.
I suggest correcting to:
```cpp
auto &entry = _APList[j];
```

Reply:
Changed.